--- index_image_mysql.php.orig	2010-04-13 17:59:08.000000000 +0200
+++ index_image_mysql.php	2010-04-13 16:03:31.000000000 +0200
@@ -30,7 +30,11 @@
 if ( !mysql_select_db( STORAGE_DB, $db ) )
     _die( "Unable to select database " . STORAGE_DB . ".\n" );
 
-$filename = ltrim( $_SERVER['REQUEST_URI'], "/");
+$filename = ltrim( $_SERVER['REQUEST_URI'], "/" );
+if ( strrpos( $filename, '?' ) !== false )
+{
+    $filename = substr( $filename, 0, strrpos( $filename, '?' ) );
+}
 
 // Fetch file metadata.
 $filePathHash = mysql_real_escape_string( $filename );
